vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/model/implementation/Serializer.java

```diff
@@ -0,0 +1,260 @@
+	}
+
+	/**
+	 * Creates a String representation of a given {@link Predicate}.
+	 * 
+	 * @see <"https://github.com/knowsys/vlog4j/wiki"> for wiki.
+	 * @param predicate a {@link Predicate}
+	 * @return String representation corresponding to a given {@link Predicate}.
+	 */
+	public static String getString(Predicate predicate) {
+		return predicate.getName() + openParenthesis + predicate.getArity() + closingParenthesis;
+	}
+
+	/**
+	 * Creates a String representation of a given {@link DataSourceDeclaration}.
+	 * 
+	 * @see <"https://github.com/knowsys/vlog4j/wiki"> for wiki.
+	 * @param dataSourceDeclaration a {@link DataSourceDeclaration}
+	 * @return String representation corresponding to a given
+	 *         {@link DataSourceDeclaration}.
+	 */
+	public static String getString(DataSourceDeclaration dataSourceDeclaration) {
+		return "@source " + dataSourceDeclaration.getPredicate().getName() + openParenthesis
+				+ dataSourceDeclaration.getPredicate().getArity() + closingParenthesis + ": "
+				+ dataSourceDeclaration.getDataSource().getSyntacticRepresentation();
+	}
+
+	/**
+	 * Creates a String representation of a given {@link Conjunction}.
+	 * 
+	 * @see <"https://github.com/knowsys/vlog4j/wiki"> for wiki.
+	 * @param conjunction a {@link Conjunction}
+	 * @return String representation corresponding to a given {@link Conjunction}.
+	 */
+	public static String getString(Conjunction<? extends Literal> conjunction) {
+		final StringBuilder stringBuilder = new StringBuilder();
+		boolean first = true;
+		for (final Literal literal : conjunction.getLiterals()) {
+			if (first) {
+				first = false;
+			} else {
+				stringBuilder.append(comma + " ");
+			}
+			stringBuilder.append(getString(literal));
+		}
+		return stringBuilder.toString();
+	}
+
+	/**
+	 * Creates a String representation corresponding to the name of a given
+	 * {@link LanguageStringConstant}.
+	 * 
+	 * @see <"https://github.com/knowsys/vlog4j/wiki"> for wiki.
+	 * @param languageStringConstant a {@link LanguageStringConstant}
+	 * @return String representation corresponding to the name of a given
+	 *         {@link LanguageStringConstant}.
+	 */
+	public static String getConstantName(LanguageStringConstant languageStringConstant) {
+		return "\"" + languageStringConstant.getString().replace("\\", "\\\\").replace("\"", "\\\"") + "\"@"
+				+ languageStringConstant.getLanguageTag();
+	}
+
+	/**
+	 * Creates a String representation corresponding to the name of a given
+	 * {@link DatatypeConstant} without an IRI.
+	 * 
+	 * @see <"https://github.com/knowsys/vlog4j/wiki"> for wiki.
+	 * @param datatypeConstant a {@link DatatypeConstant}
+	 * @return String representation corresponding to a given
+	 *         {@link DatatypeConstant}.
+	 */
+	public static String getShortConstantName(DatatypeConstant datatypeConstant) {
+		if (datatypeConstant.getDatatype().equals(PrefixDeclarations.XSD_STRING)) {
+			return "\"" + datatypeConstant.getLexicalValue() + "\"";
```

Comment:
Are we sure we don't need to escape `datatypeConstant.getLexicalValue(`) ?

Reply:
I guess if we did, the string will look different and we don't want to do that, I'm keeping the other method as well in case it was needed